### ISSUE TITLE
Feature request: String split by separator

### ISSUE THREAD

**Issue by speakk** [1bardesign/batteries#5]
Splitting a string into a table by a separator character/string is handy.

split("Hi there, mister", ",") -> { "Hi there", " mister" }

**Comment by flamendless** [comment]
Here:
```lua
local split = function(str)
	local t = {}
	for substr in string.gmatch(str, "([^/]+)") do
		table.insert(t, substr)
	end
	return t
end
```

**Comment by 1bardesign** [comment]
Addressed by 26acf1752df58c565b349cf4c6963a2d18950a36 and e75147ec77359cd31ce8c28c0a27c5b09ee5013b

Basic tests, let me know if you find any edge cases that cause problems
```lua
function test_split(s, d)
	local split_res = s:split(d)
	local join_res = table.concat(split_res, d)
	print(string.pretty(split_res))
	print(join_res)
	print("success", join_res == s)
end

local s = "a b c d e "
test_split(s, "")
test_split(s, " ")
test_split(s, " b")
test_split(s, "b c")
test_split(s, "e ")
```
result:
```
{"a", " ", "b", " ", "c", " ", "d", " ", "e", " "}
a b c d e 
success true
{"a", "b", "c", "d", "e", ""}
a b c d e 
success true
{"a", " c d e "}
a b c d e 
success true
{"a ", " d e "}
a b c d e 
success true
{"a b c d ", ""}
a b c d e 
success true
```

Am considering if i should make it return a sequence so you can chain more easily like `s:split(" "):join(", ")` but maybe it'd be better to have a specific sequence version (`split_sequence` or `splits` maybe) and keep this returning the naked table. Thoughts welcome.

Added now because I needed it today :)
